server/src/controllers/tagController.js

```diff
@@ -12,21 +12,21 @@ const addJobTags = (req, res) => {
       id: job.id
     }
   })
-    .then( (foundJob) => {
-      tagsData.forEach((tag) => {
-        Tag.findOrCreate({ 
-          where: { name: tag }
-        })
-        .spread((tag, created) => {
-          //Sequelize association that adds job and tag to JobTag join table
-          tag.addJob(foundJob)
-          res.end();
-        })
+  .then((foundJob) => {
+    tagsData.forEach((tag) => {
+      Tag.findOrCreate({ 
+        where: { name: tag }
+      })
+      .spread((tag, created) => {
+        //Sequelize association that adds job and tag to JobTag join table
+        tag.addJob(foundJob)
+        res.end();
```

Comment:
Will this end the response after the first tag is found? I feel like since tags is an array, we might need to use the [.all method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) to resolve all the promises before we end the response. Otherwise, the promise for the first tag may resolve and prematurely end the response.

Maybe I am not understanding as well, lmk tomorrow.